### DISCUSSION TITLE
Contributor Call: 4 February 2023

### DISCUSSION THREAD

**Discussion by l3xander** [freesewing/freesewing#3460] [Support] [answered]
You are all invited to the FreeSewing Contributor Call.
Please leave a comment here or [on Discord](https://discord.freesewing.org/) to add things to the agenda.
This issue will be updated after the call with notes and to-dos.

### About
FreeSewing's contributor call is an informal hangout/chat/meeting that is open to all FreeSewing contributors and users alike.
The call is organized/hosted by @tangerineshark and @kaermorhan.
The call does not have a fixed structure or agenda items. Everyone is free to suggest agenda items, ask questions, or bring ideas.

**Details**
Location: https://discord.freesewing.org (meeting room: 🔊`voice chat`)
Date: 4 February 2023 **5pm** UTC, which can be found in your local time zone [here](https://savvytime.com/converter/utc-to-belgium-antwerpen-ca-los-angeles-il-chicago-ny-new-york-city-brazil-sao-paulo-united-kingdom-london-france-paris-south-africa-johannesburg-msd-india-mumbai-hkt-japan-tokyo-australia-sydney/feb-4-2023/5pm).
Or check the Events tab in Discord!
The meetings will always be at 6pm, Antwerp time.

### Agenda

- Documentation and style guide (@BenJamesBen can explain)
- Consolidating *annotation* plugins (@joostdecock can explain)
- Killing `@freesewing/i18n` in favor of sidecar translations (@joostdecock can explain)
- React components shall use names exports and `.mjs` extension, in line with the changes in our other code
- clone vs copy in the core API and how to have it both ways (@eriese @BenJamesBen @anna-puk and @woutervdub discussed this)
> Notes will be added after the call.

**Comment by SeaZeeZee** [comment]
Documentation process and style guide issues

**Reply by BenJamesBen** [reply]
Style Guide background/info:
https://github.com/freesewing/freesewing/issues/3389
https://freesewing-dev-website-3kotiidq7-benjamesben.vercel.app/guides/writers/style

The above Vercel link also contains a draft of the Writers Guide documentation that I have been working on. Background:
https://github.com/freesewing/freesewing/issues/3385

**Comment by BenJamesBen** [comment]
For the Pattern design tutorial, what voice/pronouns should be used: "you" or "we"?
See:
https://github.com/freesewing/freesewing/pull/3391
https://github.com/freesewing/freesewing/issues/2982

**Comment by joostdecock** [comment, marked as answer]
# Call notes

We have some new people on the call, including `lizzomatic`, who is dabbling into parametric design themself but in Python.
A bit later, `Martha` joined but they preferred to remain silent for now.

Obviously we are all very happy to have these new people joining us, and they are very welcome.

There's a small question on how to pronounce Joost's name. It rhymes with `ghost`.

## Documentation and style guide

@BenJamesBen  (BJB) explains the use of we/our vs you/your in the pattern design tutorial.
There used to be inconsistent wording (a mix between both) and BJB changed it
all to we/our.  But than @SeaZeeZee was concerned that you/your is not a good use.
They feel that you/yours is more activating, putting the person reading the
tutorial in control. Rather than the feeling that it's something the writer is
doing.

BJB explains that the change was made for consistency since there was a mix between you and we.

@joostdecock chimes in to explain that historically, everything used `you`. However, after reading a book about writing good documentation, the tutorial was switched to `we` in line with the advice in that book.
The idea is that for this sort of documentation (tutorials) one takes the reader by the hand to go through it together. Hence `we`.

There's a consensus that for tutorials, we use **we**, for reference and other
documentation, we use **you**.

If we're lucky, BJB might write it in the style guide :)

## Consolidating annotation plugins

The idea is to consolidate the various plugins that annotate the pattern (grainline, dimensions and so on) into one plugin.

Also to add is the crossbox that Wouter added to some of his patterns.

There's a consensus that this is a good idea, so we'll add it to our todo list for v3: #3517

## i18n remake to a sidecar approach

There's some discussion on the topic, mostly between Eriese and joost.
Consensus is that the sidecar pattern is the way to go. There's still some
debate on whether it's worthwhile to use a named export for namespaces. Joost
says that he'd like to kick the tires a bit to see if it's worthwhile with
regards to the initial bundle size.

## React components .mjs and named exports

Joost is working to make all react components .mjs with named exports. In line
with the ESM-only approach in the core library.

This is a wip, but will be merged at some point when people will have to
re-sync.

## Clone vs Copy in API methods

There's a request to have an extra parameter to make methods like `Point.shift()` and others use `Point.clone()` under the hood, rather than `Point.copy()`. This would mean all their attributes are moved over (like text and so on).

Enoch explains that it would be good if it's possible to force clone behaviour instead of copy.
He'd like to add a parameter for this.

Everybody likes it as an idea, but Joost prefers to make this explicit by having a different _clone_ method. Under the hood, it can call the method with the extra parameter. But it will make the API more discoverable if there's an explicit method for it.

This reasoning initially finds common support, but then people start getting worried about this adding too much documentation. There's some muttering on the overheads of maintaining all that documentation, but in the end everyone agreed to add the extra methods.

The only thing that's not yet clear is how they should be named. Something like `shiftClone` or rather `cloneShift`.
To be decided later.

**Comment by joostdecock** [comment]
Closing to increase visibility of active discussions
